### DISCUSSION TITLE
Seeking Clarification for JVM Options and Flags

### DISCUSSION THREAD

**Discussion by bibiqt40** [itzg/docker-minecraft-server#2561] [General]
Hello! New Docker user here. Have spent the last few days labbing a container server inbetween food comas.

I have yet to bring my bare metal server live with a Linux environment (Ubuntu) and install docker and bring my yaml file up, but so far I've been labbing everything with VitualBox and an Ubuntu image.

I've brought a vanilla server online and with minimal RAM its been fine for a proof of concept, I've plans to mod this with 5-6 FORGE Mods, however I'm trying to nail down the other bits and bobs for the container.

**My question is this:**
I've seen some dicussions threads talking about JVM Options and Aikar's Flags and the various RAM settings you're meant to allocate (i.e. I will be allocating 12 with the environment tag MEMORY: "12G"), however when it comes to GC and min/max RAM for this image do I need to:

a) Only include the following tags in my yaml as outlined by the ["Enabling Aikar's flags"](https://docker-minecraft-server.readthedocs.io/en/latest/configuration/jvm-options/#enable-aikars-flags) wiki:

**environment:**
_MEMORY: "12G"
USE_AIKAR_FLAGS= "TRUE"_

b) Include the entire copy and paste as is found on [Aikar's blog post](https://aikar.co/2018/07/02/tuning-the-jvm-g1gc-garbage-collector-flags-for-minecraft/) as well as the tags

**environment:**
_MEMORY: "12G"
USE_AIKAR_FLAGS= "TRUE"_

Apologies for probably rehashing a topic thats been talked about and talked about, please let me know if there was anything else I should include.

Cheers

**Comment by itzg** [comment]
Just option a is needed. Set `USE_AIKAR_FLAGS` to "TRUE" and the container include all of the right JVM options for the memory amount being requested.
